### PR TITLE
Add Client option for changing gNMI request dump log level

### PR DESCRIPTION
**Pull request by wenovus**
This is very verbose if there are a lot of queries happening, so hardcoding level 1 is not appropriate.

### CODE REVIEW

**Comment by wenovus** [comment]
Daniel also suggested increasing above 2, which I'm also in favour of -- @greg-dennis @DanG100 are you aware of any guidelines on how levels are set besides an arbitrary number?

**Comment by greg-dennis** [comment]
@wenovus I'm not aware of any general guidelines. I think that determination is made by the project itself. Here is what we document in Ondatra, and it would be nice if we can attain for some self-consistency across  OpenConfig projects: https://github.com/openconfig/ondatra#logging-verbosity

**Comment by wenovus** [comment]
> @wenovus I'm not aware of any general guidelines. I think that determination is made by the project itself. Here is what we document in Ondatra, and it would be nice if we can attain for some self-consistency across OpenConfig projects: https://github.com/openconfig/ondatra#logging-verbosity

@DanG100 How about having a new field in ygnmi.Client where you can set the logging level for dumps? Let it default to 1? I'm wary of changing from what Ondatra is already using.

I question whether it is worth trying to create a consistent logging level across OpenConfig projects just because different things might be of different importance to each project. In this case we'd have to change Ondatra's existing logging verbosity guidelines if we want `SetRequest` dumps to have different levels than the other information types in the table you referenced. In the future, we might decide in lemming that it is yet too verbose and then need another level, which would break user expectations again.

**Comment by DanG100** [comment]
1. I think doing prototext.Format and having log message span multiple lines makes filtering the log messages harder
2. In lemming multiple setrequests are happening every second, whereas ondatra tests typical have much fewer

Some kind of client option for logging seems reasonable

**Comment by wenovus** [comment]
@DanG100 ping on review, PTAL

File: ygnmi/ygnmi.go [hunk 1]
```diff
@@ -127,8 +127,9 @@ func (v *Value[T]) String() string {
 
 // Client is used to perform gNMI requests.
 type Client struct {
-	gnmiC  gpb.GNMIClient
-	target string
+	gnmiC               gpb.GNMIClient
+	target              string
+	requestDumpLogLevel log.Level
```
Comment (greg-dennis) [inline ygnmi/ygnmi.go:132]: "requestLogLevel" may be sufficiently descriptive?

File: ygnmi/gnmi.go [hunk 2]
```diff
@@ -73,7 +73,7 @@ func subscribe[T any](ctx context.Context, c *Client, q AnyQuery[T], mode gpb.Su
 			dt = gpb.GetRequest_STATE
 		}
 		sub = &getSubscriber{
-			client:   c.gnmiC,
+			client:   c,
```
Comment (greg-dennis) [inline ygnmi/gnmi.go:76]: A little torn between whether it's worth storing a reference to the full client here or whether the gnmi client and logging level should be two fields. Daniel can make the call :)

**Review by greg-dennis — APPROVED** [review]
A couple optional things. Daniel should make the final call.

File: ygnmi/ygnmi.go (hunk 1 above)
Comment (DanG100) [reply ygnmi/ygnmi.go:132]: +1

File: ygnmi/gnmi.go (hunk 2 above)
Comment (DanG100) [reply ygnmi/gnmi.go:76]: i don't have a strong opinion on this

Comment (wenovus) [reply ygnmi/gnmi.go:76]: I'll keep it as is then. This is an unexported field and we can modify this later.

**Review by DanG100 — APPROVED** [review]

**Pull request merged**